src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/Schema/XamlNamespace.cs

```diff
@@ -98,43 +90,45 @@ private XamlType TryGetXamlType(string typeName)
 
             // And save it in our cache
             xamlType = SchemaContext.GetXamlType(type);
-            if (xamlType != null)
+            if (xamlType == null)
             {
-                xamlType = XamlSchemaContext.TryAdd(_typeCache, typeName, xamlType);
+                return null;
             }
+
+            xamlType = XamlSchemaContext.TryAdd(_typeCache, typeName, xamlType);
             return xamlType;
         }
 
         private XamlType TryGetXamlType(string typeName, Type[] typeArgs)
         {
-            // Can't get an array of open generic and then call MakeGenericType on it.
-            // So we need to process subscripts on generics ourselves.
+            Debug.Assert(typeArgs.Length > 0, "This method should only be called for generic types.");
+
+            // It is not possible to get an array of open generic and then call
+            // MakeGenericType on it so we need to process array subscripts.
             string subscript;
             typeName = GenericTypeNameScanner.StripSubscript(typeName, out subscript);
-            typeName = MangleGenericTypeName(typeName, typeArgs.Length);
+            typeName = typeName + KnownStrings.GraveQuote + typeArgs.Length;
```

Comment:
i love all these code cleanup changes! this is probably one i'd prefer to keep the MangleGenericTypeName method, as it is easier to read and get an understanding of what `typeName + KnownStrings.GraveQuote + typeArgs.Length` is doing. up to you though

Reply:
Sure. Fixed